packages/driver/cypress/e2e/e2e/origin/cookie_login.cy.ts

```diff
@@ -1,95 +1,629 @@
+      ◦ else
+        • displays "Welcome, <username>"
+  ****************************************************************************/
+
+  describe('general behavior', () => {
+    let username
+
+    beforeEach(() => {
+      username = getUsername()
+    })
+
+    it('works in a session', () => {
+      cy.session(username, () => {
+        cy.visit('/fixtures/primary-origin.html')
+        cy.get('[data-cy="cookie-login"]').click()
+        cy.origin('http://foobar.com:3500', { args: { username } }, ({ username }) => {
+          cy.get('[data-cy="username"]').type(username)
+          cy.get('[data-cy="login"]').click()
+        })
+
+        cy.getCookie('user').its('value').should('equal', username)
+      }, {
+        validate () {
+          cy.getCookie('user').its('value').should('equal', username)
+        },
+      })
+
+      cy.visit('/welcome')
+      verifyLoggedIn(username)
+    })
+
+    // https://github.com/cypress-io/cypress/issues/21201
+    it('works with aliased localhost', () => {
+      cy.visit('/fixtures/primary-origin.html')
+      cy.get('[data-cy="cookie-login-alias"]').click()
+
+      cy.origin('http://foobar.com:3500', { args: { username } }, ({ username }) => {
+        cy.get('[data-cy="username"]').type(username)
+        cy.get('[data-cy="login"]').click()
+      })
+
+      verifyLoggedIn(username)
+    })
+
+    it('makes cross-origin cookies readable via document.cookie', () => {
       cy.visit('/fixtures/primary-origin.html')
       cy.get('[data-cy="cookie-login"]').click()
-      cy.origin('http://foobar.com:3500', () => {
-        cy.get('[data-cy="username"]').type('ZJohnson')
+      cy.origin('http://foobar.com:3500', { args: { username } }, ({ username }) => {
+        cy.get('[data-cy="username"]').type(username)
         cy.get('[data-cy="login"]').click()
       })
-    }, {
-      validate () {
-        cy.getCookie('user').its('value').should('equal', 'ZJohnson')
-      },
+
+      cy.document().its('cookie').should('equal', `user=${username}`)
     })
 
-    cy.visit('/welcome')
-    verifyLoggedIn('ZJohnson')
+    it('cy.clearCookie() -> logged out', () => {
+      cy.visit('/fixtures/primary-origin.html')
+      cy.get('[data-cy="cookie-login"]').click()
+      cy.origin('http://foobar.com:3500', { args: { username } }, ({ username }) => {
+        cy.get('[data-cy="username"]').type(username)
+        cy.get('[data-cy="login"]').click()
+      })
+
+      cy.clearCookie('user')
+      cy.reload()
+      verifyLocalhostNotLoggedIn()
+    })
+
+    it('cy.clearCookies() -> logged out', () => {
+      cy.visit('/fixtures/primary-origin.html')
+      cy.get('[data-cy="cookie-login"]').click()
+      cy.origin('http://foobar.com:3500', { args: { username } }, ({ username }) => {
+        cy.get('[data-cy="username"]').type(username)
+        cy.get('[data-cy="login"]').click()
+      })
+
+      cy.clearCookies()
+      cy.reload()
+      verifyLocalhostNotLoggedIn()
+    })
```

Comment:
Is this test a duplicate?

Reply:
A duplicate of the one above it? That covers _cy.clearCookie()_ (clearing a single cookie) and this one covers _cy.clearCookie**s**()_ (clearing all cookies).

Reply:
I missed the lone `s`! 😭